cloudevents/abstract/event.py

```diff
@@ -0,0 +1,97 @@
+import abc
+import typing
+from typing import TypeVar
+
+
+class CloudEvent(abc.ABC):
+    """
+    Used internally for multi-framework event type integration
+    """
+
+    @classmethod
+    @abc.abstractmethod
+    def create(
+        cls,
+        attributes: typing.Dict[str, typing.Any],
+        data: typing.Optional[typing.Any],
+    ) -> "CloudEvent":
+        """
+        Factory function.
+        We SHOULD NOT use the __init__ function as the factory because
+        we MUST NOT assume how the __init__ function looks across different python
+        frameworks
+        """
+        pass
+
+    @property
+    @abc.abstractmethod
+    def _attributes_read_model(self) -> typing.Dict[str, typing.Any]:
+        """
+        :return: attributes of this event
+
+        you MUST NOT mutate this dict
+        implementation MAY assume the dic will not be mutated
+        """
+        pass
```

Comment:
Why don't we go with public `attributes` property? This way each implementation will expose the attributes which may be beneficial.

Reply:
it is to prevent confision. Documented in the function this reason

Reply:
What kind of confusion? We're basically talking about having an interface that exposes desired public APIs.

I'd say it is beneficial to talk it through which properties must be exposed and which not.

Reply:
* What is the difference between `event.get("myattr")` and
        `event.attributes.get("myattr")`
        * What SHOULD I use `event["myattr"]` or `event.attributes["myattr"]` ?

Reply:
I'm not saying that one should not use .get way. I'm saying that it may be beneficial to expose attributes sometimes.

Reply:
OK fine with me